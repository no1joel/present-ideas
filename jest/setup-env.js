// test/setup-env.js
// add this to your setupFilesAfterEnv config in jest so it's imported for every test file
import fetch, { Headers, Request, Response } from "node-fetch";
import { TextDecoder as UtilTextDecoder } from "util";
import { server } from "../mock_server";

if (!globalThis.fetch) {
  // @ts-ignore
  globalThis.fetch = fetch;
  globalThis.Headers = Headers;
  // @ts-ignore
  globalThis.Request = Request;
  // @ts-ignore
  globalThis.Response = Response;
}

if (!globalThis.TextDecoder) {
  globalThis.TextDecoder = UtilTextDecoder;
}

beforeAll(() => server.listen());
// if you need to add a handler after calling setupServer for some specific test
// this will remove that handler for the rest of them
// (which is important for test isolation):
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
