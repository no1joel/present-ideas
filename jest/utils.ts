import {
  render as testingLibraryRender,
  RenderOptions,
} from "@testing-library/vue";
import linkify from "vue-linkify";
import { createMemoryHistory, createRouter } from "vue-router";
import routes from "../present_ideas/present_ideas/static/router/routes";
import getStore from "../present_ideas/present_ideas/static/store";

const mockRouter = createRouter({ routes, history: createMemoryHistory() });

export const render = async (component, options = {}) => {
  const store = getStore();
  const defaultRenderOptions: RenderOptions = {
    global: {
      plugins: [store, mockRouter],
      directives: { linkified: linkify },
    },
  };
  const renderOptionsWithDefaults = { ...defaultRenderOptions, ...options };

  const result = testingLibraryRender(component, renderOptionsWithDefaults);

  return result;
};
