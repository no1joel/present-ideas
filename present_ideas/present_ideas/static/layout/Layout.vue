<template>
  <div>
    <TopNav />
    <slot />
  </div>
</template>

<script>
import TopNav from "../components/TopNav.vue"
export default {
  components: { TopNav },
}
</script>