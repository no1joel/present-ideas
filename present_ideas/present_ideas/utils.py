from typing import Iterable, List

from sheets.spreadsheet import (
    add_row,
    delete_row,
    get_rows,
    get_worksheets,
    set_cell_value,
)


def check_name(name: str) -> bool:
    """Check if the provided name is a sheet."""

    worksheet_names = get_all_names()
    return name.lower() in worksheet_names


def get_all_names() -> Iterable[str]:
    """Return people's names."""

    return (
        worksheet.title.lower()
        for worksheet in get_worksheets()
        if "template" not in worksheet.title.lower()
    )


def get_present_ideas(their_name: str) -> List[dict]:
    # TODO: Embed urls
    _header, *rows = get_rows(their_name)
    presents = [{**row, "index": index} for index, row in enumerate(rows)]
    return presents


def get_row_index(thing_index: int) -> int:
    # TODO: Why is it plus 3?!
    return thing_index + 3
