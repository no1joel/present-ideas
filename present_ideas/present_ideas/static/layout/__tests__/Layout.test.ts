import { render } from "../../../../../jest/utils";
import Layout from "../Layout.vue";

describe("Layout", () => {
  it("matches snapshot", () => {
    expect(render(Layout).container).toMatchSnapshot();
  });
});
