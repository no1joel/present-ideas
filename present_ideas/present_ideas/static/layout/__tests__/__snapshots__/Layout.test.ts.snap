// Vitest Snapshot v1

exports[`Layout > matches snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Present Ideas
      </a>
      <!--v-if-->
    </nav>
    <div
      class="mt-3"
    >
      
      
    </div>
  </div>
</div>
`;
